schema/schema.yaml

```
@@ -1,48 +1,24 @@
 $schema: https://fiboa.github.io/schema/v0.1.0/schema.json
 required:
-  - ml:creation_method
   - ml:boundary_type
+  - ml:creation_date
   - ml:verified_from
-  - ml:verified_until
   - ml:author
-  - ml:quality
-  - ml:machine_generated
   - ml:country_code
-  - ml:admin0
   - ml:admin1
   - ml:admin2
-  - ml:crop_category
 properties:
-  ml:creation_method:
-    type: string
-  ml:creation_method_detail:
-    type: string
   ml:ml:boundary_type:
     type: string
-    required: true
   ml:creation_date:
     type: date-time
   ml:verified_from:
     type: date-time
-  ml:verified_until:
-    type: date-time
   ml:author:
     type: string
-  ml:quality:
-    type: string
-  ml:confidence:
-    type: float
-  ml:machine_generated:
-    type: boolean
   ml:country_code:
     type: string
-  ml:admin0:
-    type: string
   ml:admin1:
     type: string
```

```suggestion
    type: string
    minLength: 5
    maxLength: 7
    pattern: ^[A-Z]{3}-[A-Z0-9]{1,3}$
```

I believe that's correct?

---

I believe `^[A-Z]{2}-[A-Z0-9]{1,3}$` is the right format.

---

Uh, that's confusing. ml:country_code is using a three letter code. Shouldn't this be aligned?

---

We could – there are some more peculiarities involving the codes, specifically HASC_2. I'm looking into it right now, and I'll follow up here once I get through them.

---

If we were to cover all anomalies in the HASC 2 codes, the pattern would roughly look like this:
`^[A-Z]{1,2}\.[A-Z]{2,5}(?:\.[A-Z0-9]{1,2})$`

Please excuse my naïve regex, I meant to convey that the third part of the code is nonexistent in some cases.
I think we should unify the first two parts with ISO 3166-2 codes, but we are reliant are HASC 2 for the third part. Since we have regions and countries that do not have this third part, I suggest we may want to make `ml:admin2` optional. What do you think?

---

Could you provide some examples of these codes?

I find it a bit weird that country_code, admin1 and admin2 are all differently formatted...

---

I agree; I think I would prefer unifying the format of the 3 features.
Some HASC 2 examples:
Most standard: `ZW.MI.GW`
Some edge cases:
`A.OK.KP`
`KG.WATER`
`LV.DO` – Around 550 of 30,000 are in this format
`CM.CE.M`
`SK.KI.K2`

---

Like I briefly mentioned above, I think it might be best for `admin2` to be _optional_ and look something like this:
`^[A-Z]{3}-[A-Z0-9]{1,3}-[A-Z0-9]{0,2}$`, where the first and second part are identical to `ml:country_code` and `ml:admin1` respectively

---

Thanks, I agree! The last part could be longer than two characters, right?

---

I believe 2 characters is the longest (valid) case in HASC 2.

---

Okay, great. Then `^[A-Z]{3}-[A-Z0-9]{1,3}-[A-Z0-9]{0,2}$` should work.

---

Although, why do we duplicate the first two parts?

Would it make sense to just provide the last part?

---

Yes, I believe so. I did change it to `^[A-Z]{3}-[A-Z0-9]{1,3}-[A-Z0-9]{1,2}$`, since if the last was at 0 length they shouldn't include it.

---

That's also a good point – I left it in there in case people would want the full code without having to merge it themselves. Let me talk to the team and see what they think.

---

We decided to go with not duplicating the other features to reduce redundancy.

---

Sounds good to me, thanks.